gnovm/memfile.go

```diff
@@ -87,6 +89,25 @@ func (mempkg *MemPackage) Validate() error {
 		prev = file.Name
 	}
 
+	pIndex := strings.Index(mempkg.Path, "/p/")
+	if pIndex > 0 && strings.Count(mempkg.Path[:pIndex], "/") == 0 {
+		for _, file := range mempkg.Files {
+			fset := token.NewFileSet()
+			astFile, err := parser.ParseFile(fset, file.Name, file.Body, parser.ImportsOnly)
+			if err != nil {
+				continue // can be other files like LICENSE, README or empty gno files
+			}
+			for _, imp := range astFile.Imports {
+				// ensure the pkg is a realm by checking if the path contains /r/ and no other / character before it (i.e protect from gno.land/p/demo/r/)
```

Comment:
maybe we should add a test for the case `gno.land/p/demo/r/`

Reply:
7b64799